lua/rocks-dev/local-rock-handler.lua

```diff
@@ -19,18 +19,26 @@ function rock_handler.get_sync_callback(rock)
         ---@param report_progress fun(message: string)
         ---@param report_error fun(message: string)
         return nio.create(function(report_progress, report_error)
+            local event = nio.control.event()
```

Comment:
suggestion: perhaps `nio.conttol.future()` would be cleaner here?

Then you could call

```lua
local remove_local_rock = future.wait()
```

It's more loc, but less stateful :)

Reply:
Good suggestion, I like that with future you understand what is being waited for better